enterprise/reporting/reporting.go

```diff
@@ -43,18 +43,21 @@ const (
 )
 
 type HandleT struct {
-	clients                       map[string]*types.Client
-	clientsMapLock                sync.RWMutex
-	logger                        logger.Logger
-	reportingServiceURL           string
-	namespace                     string
-	workspaceID                   string
-	instanceID                    string
-	workspaceIDForSourceIDMap     map[string]string
-	workspaceIDForSourceIDMapLock sync.RWMutex
-	whActionsOnly                 bool
-	sleepInterval                 time.Duration
-	mainLoopSleepInterval         time.Duration
+	init                      chan struct{}
+	onceInit                  sync.Once
+	clients                   map[string]*types.Client
+	clientsMapLock            sync.RWMutex
+	logger                    logger.Logger
+	reportingServiceURL       string
+	namespace                 string
+	workspaceID               string
+	instanceID                string
+	workspaceIDForSourceIDMap map[string]string
+	piiReportingSettings      map[string]bool
+	configSubscriberLock      sync.RWMutex
```

Comment:
since we are not modifying maps, but replacing them when a new configuration is received this lock is not really needed, right?

Reply:
We have added a lock since we are modifying the map here - https://github.com/rudderlabs/rudder-server/pull/2655/files#diff-5639a722fc70b6a04ec6daf63094d6bb127454dd8587e1b3466785305e515c3eR120

Reply:
From what I can see, we are not modifying the map there, we are assigning a new map to the field, two different things

Reply:
Removed.